libs/ts-rest/core/src/lib/client.ts

```diff
@@ -140,12 +155,33 @@ export type AppRouteFunction<
       args: Prettify<DataReturnArgs<TRoute, TClientArgs>>
     ) => Promise<Prettify<ApiRouteResponse<TRoute['responses']>>>;
 
+/**
+ * Returned from a mutation or query call when NoUnknownStatus mode is enabled
+ */
+export type AppRouteFunctionNoUnknownStatus<
+  TRoute extends AppRoute,
+  TClientArgs extends ClientArgs
+> = AreAllPropertiesOptional<DataReturnArgs<TRoute, TClientArgs>> extends true
+  ? (
+      args?: Prettify<DataReturnArgs<TRoute, TClientArgs>>
+    ) => Promise<Prettify<ApiRouteResponseNoUnknownStatus<TRoute['responses']>>>
+  : (
+      args: Prettify<DataReturnArgs<TRoute, TClientArgs>>
+    ) => Promise<
+      Prettify<ApiRouteResponseNoUnknownStatus<TRoute['responses']>>
+    >;
+
 export interface ClientArgs {
   baseUrl: string;
   baseHeaders: Record<string, string>;
   api?: ApiFetcher;
   credentials?: RequestCredentials;
   jsonQuery?: boolean;
+  /**
+   * Ensures that the responses from the server match those defined in the
+   * contract.
+   */
+  throwOnUnknownStatus?: boolean;
```

Comment:
This should be removed from this interface as it is being used by the rest of the libraries